### PR TITLE
feat(grid): adding image column data type

### LINKED ISSUE #12237: igx-grid doesn't have filtering on with allow-filter="true" and incorrect value for filter-mode
## Description  
I create an `igc-grid` with `allow-filtering="true" filter-mode="excelStyleFiltering"`. The correct filter-mode value is `excelStyleFilter`. Filtering is not on with any mode.

 * igniteui-angular version: 14.2.x
 * browser: all

## Steps to reproduce  

1. Create an `igc-grid` with `allow-filtering="true" filter-mode="excelStyleFiltering"`.
2. Run the app

## Result  
Filtering is not on.

## Expected result  
Filtering should be on, since `allow-filtering="true"`.

### LINKED ISSUE #12255: igx-grid column dataType="image"
## Is your feature request related to a problem? Please describe.
I would like to have default handling for an image column. 

## Describe the solution you'd like
I should be able to define the following type of column:

```html
<igx-column dataType="image" field="imageSourcesField"></igx-column>
```

The grid should disable filtering, sorting, grouping, editing and summaries for this column by default, but should allow for custom strategies to be provided, if the user knows how to sort/filter/group the image column.

## Describe alternatives you've considered
Currently, I can pass a template, which renders the images, but I also have to turn off all the rest of the features for the column manually.

## Additional context
This type of column is just for ease-of-use. It can be achieved with the available features, but this requires more configuration on the developer side.

### PR DESCRIPTION
**Pull request by kdinev**
Closes #12255 
Closes #12237 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [x] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/columns/column.component.ts [hunk 1]
```diff
@@ -1850,10 +1849,14 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy, ColumnTy
                 case GridColumnDataType.DateTime:
                     this.filters = IgxDateTimeFilteringOperand.instance();
                     break;
+                case GridColumnDataType.Image:
+                    this.filterable = false;
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/grids/columns/column.component.ts:1853]: As a user, I am able to make the image column sortable, but this line prevents me from setting it as filterable, not sure if expected.

Comment (kdinev) [reply projects/igniteui-angular/src/lib/grids/columns/column.component.ts:1853]: Expand the lines above and you'll see this is executed only if there's no filters provided.

**Review by zdrawku — CHANGES_REQUESTED** [review]
@kdinev, spec and changes looks good. Something else that we have to cover is the Tree Grid Cell and Expandable cell (master-detail).

- igx-expandable-grid-cell

image
 
- igx-tree-grid-cell
image

**Comment by zdrawku** [comment]
Perfect:


image

image

**Review by zdrawku — DISMISSED** [review]

**Comment by kdinev** [comment]
@zdrawku @hanastasov I know the changes would now seem insane for this feature, but I talked to @ChronosSF and I'm pushing some test stabilization changes to the same PR.

**Review by hanastasov — APPROVED** [review]

**Comment by hanastasov** [comment]
Filtering is also fine :)
image

**Pull request merged**